src/components/tab-panel/tab-panel.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+    Component,
+    Element,
+    h,
+    Prop,
+    EventEmitter,
+    Event,
+    Host,
+} from '@stencil/core';
+import { Tab } from '../tab-bar/tab.types';
+
+/**
+ * @slot - Content to put inside the `tab-panel`. Each slotted element must have the `id` attribute equal to the id of the tab it belongs to.
+ */
+@Component({
+    tag: 'limel-tab-panel',
+    shadow: true,
+})
+export class TabPanel {
+    @Prop({ mutable: true })
+    public tabs: Tab[] = [];
+
+    @Element()
+    private host: HTMLElement;
+
+    @Event()
+    protected changeTab: EventEmitter<Tab>;
```

Comment:
I don't really understand how this manages to pass the compiler… 🤔

According to [this answer on StackOverflow](), `protected` works like this in typescript:

> The **protected** modifier acts much like the **private** modifier with the exception that members declared **protected** can also be accessed within deriving classes.

I don't really know what a *deriving class* is in typescript, but unless the definition includes "stuff in the same folder" or "stuff with the same prefix", I really can't figure out how the compiler determines that `TabPanelContentExample` is a "deriving class" of `TabPanel`…? 😂 

Regardless, it works, so I'm fine with it 😄

Reply:
Hmm :thinking: `TabPanelContentExample` is not derived from `TabPanel` so thats a bit weird haha. I think deriving class is probably just a sub-class of another :slightly_smiling_face: 

Maybe we should document this in some way... we should not have to declare this at all since the event comes from the `tab-bar`, but the compiler complains that "there is no such event" if you specify the handler on `tab-panel` like in our example component. It's still valid though since all events propagates through the DOM.

I had it as `private` first, but then I got warnings about an unused property. I'm not sure why I did not have it as `public` though :thinking: Maybe I didn't want it to end up in the documentation, but now I see that it's there anyway

Reply:
Oh, yeah, this isn't even the one used on [line 92 in tab-panel-content.tsx](https://github.com/Lundalogik/lime-elements/pull/627/files?file-filters%5B%5D=.js&file-filters%5B%5D=.mdx&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-68164d3e9ca96b930b2967af27e2561cR92)… I just searched the diff for a .emit, and that was the only one I found so I didn't really look further into it.

Ok, but that explains why `protected` works here 😄

So, the compiler complains about [line 41 in tab-panel.tsx](https://github.com/Lundalogik/lime-elements/pull/627/files?file-filters%5B%5D=.js&file-filters%5B%5D=.mdx&file-filters%5B%5D=.ts&file-filters%5B%5D=.tsx#diff-910770ff727a8a4c863417a2b6566e32R41) unless this is declared here?

Yeah, that's annoying… 😖

On the other hand, since the "normal" way of consuming `tab-bar` will be through this component, it doesn't seem wildly inappropriate that this component documents the `changeTab` event, does it? From the perspective of someone reading the docs I mean.

Reply:
No I guess not :slightly_smiling_face: It could just as well have been emitted from this component :slightly_smiling_face: